src/remote/entrypoints.py

```diff
@@ -0,0 +1,234 @@
+    if not (config.root / ".git").exists():
+        return
+
+    # make sure we don't keep adding to .gitignore
+    gitignore = config.root / ".gitignore"
+    if gitignore.exists():
+        for line in gitignore.read_text().splitlines():
+            if line.startswith(".remote"):
+                return
+
+    with gitignore.open("a") as f:
+        f.write("\n")
+        f.write(".remote*")
+        f.write("\n")
+
+    print("Added '.remote*' to .gitignore")
+
+
+@log_exceptions
+def remote_ignore():
+    parser = argparse.ArgumentParser(description="Add more patterns to the ignores list")
+    parser.add_argument("ignore", type=str, nargs="+", help="a pattern describing a file to ignore in rsync format")
+    args = parser.parse_args()
+
+    config = load_cwd_workspace_config()
+    config.ignores.add_ignores(args.ignore)
+    save_config(config)
+
+
+@log_exceptions
+def remote_host():
+    """Print the remote host in use and exit"""
+    workspace = SyncedWorkspace.from_cwd()
+    print(workspace.remote.host)
+
+
+@log_exceptions
+def remote_set():
+    parser = argparse.ArgumentParser(description="Set a new default remote host for the workspace")
+    parser.add_argument("index", type=int, help="an index of host in config file to use by default (strating from 1)")
+    args = parser.parse_args()
+
+    config = load_cwd_workspace_config()
+    if len(config.configurations) < args.index:
+        print(f"Index is too big ({args.index}). Only have {len(config.configurations)} hosts to choose from.")
+        sys.exit(1)
+    elif args.index < 1:
+        print(f"Index should be 1 or higher")
+        sys.exit(1)
+    # we use 0-base index internally
+    index = args.index - 1
+    config.default_configuration = index
+    save_config(config)
+
+    print(f"Remote host is set to {config.configurations[index].host}")
+
+
+@log_exceptions
+def remote():
+    parser = argparse.ArgumentParser(
+        description="Sync local workspace with remote directory and execute the command remotely"
+    )
+    parser.add_argument("-n", "--dry-run", action="store_true", help="do a dry run of a pull")
+    parser.add_argument("-m", "--mirror", action="store_true", help="mirror local files on remote host")
+    parser.add_argument("-v", "--verbose", action="store_true", help="increase verbosity")
+    parser.add_argument(
+        "-e", action="store_true", help="(deprecated) no effect in the new implementation",
+    )
+    parser.add_argument("command", type=str, nargs=argparse.REMAINDER, help="a command to execute")
+    args = parser.parse_args()
+
+    if not args.command:
+        print("Missing command")
+        sys.exit(1)
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO, format=BASE_LOGGING_FORMAT)
+
+    workspace = SyncedWorkspace.from_cwd()
+    workspace.push(dry_run=args.dry_run, verbose=args.verbose, mirror=args.mirror)
+    workspace.execute(args.command, dry_run=args.dry_run)
```

Comment:
What happens if an exception is raised by .execute ... does pull get executed? 
I changed the behavior of bash remote recently because I found that users actually expect that even if the command fails (e.g. gradle test failed), users expect files (e.g. test output) to get synced back so they can inspect them locally.